src/citation/views/citation_entry_view.py

```diff
@@ -141,6 +149,31 @@ def doi_search(self, request):
         result = open_alex.get_data_from_doi(doi_string)
         return Response(result, status=200)
 
+    @action(detail=False, methods=["get"], permission_classes=[IsAuthenticated])
+    def url_search(self, request):
+        try:
+            url_string = request.query_params.get("url", None)
+            scraper_result = cloudscraper.create_scraper().get(url_string)
+            status_code = scraper_result.status_code
+            dois = []
+
+            if status_code >= 200 and status_code < 400:
+                content = BeautifulSoup(scraper_result.content, "lxml")
+                dois = re.findall(
+                    r"10.\d{4,9}\/[-._;()\/:a-zA-Z0-9]+?(?=[\";%<>\?#&])", str(content)
```

Comment:
I've seen DOI regex used in a bunch of locations. Can we store it externally somewhere in a constant?